backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/DeleteByQueryPerformer.java

```diff
@@ -43,17 +43,21 @@
     private final Client client;
     private final ExecutorService executor;
     private final int batchSize;
+    private final String indexName;
+    private final String typeName;
 
     @Inject
-    public DeleteByQueryPerformer(Client client, @Named("AsyncExecutor") ExecutorService executor) {
-        this(client, executor, DEFAULT_BATCH_SIZE);
+    public DeleteByQueryPerformer(Client client, @Named("AsyncExecutor") ExecutorService executor, String indexName, String typeName) {
```

Comment:
Can you define custom type to differentiate indexName and typeName by strong typing?